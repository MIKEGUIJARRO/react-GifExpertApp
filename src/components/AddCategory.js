import React, { useState } from "react";
import PropTypes from "prop-types";
//Aqui recibimos las (props) de AddCategory 
export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState("");

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setInputValue("");
            setCategories((items) => {
                return [inputValue, ...items];
            });
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input type="text" value={inputValue} onChange={handleInputChange} />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}
