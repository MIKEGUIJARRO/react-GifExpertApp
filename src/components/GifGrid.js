import React, { Fragment } from "react";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <Fragment>
            <h3>{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading ...</p>}
            <div className="card-grid">
                {
                    images.map((image) => {
                        return <GifGridItem key={image.id} {...image} />
                    })
                }
            </div>
        </Fragment>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}