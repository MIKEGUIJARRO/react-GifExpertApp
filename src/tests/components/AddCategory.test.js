import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { AddCategory } from "../../components/AddCategory";


describe("<AddCategory /> test", () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    //Lifecycle test proofs
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test("Shows the <AddCategory /> correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Should change the textbox", () => {
        const input = wrapper.find("input");
        const value = "Hola mundo desde change textbox";
        input.simulate("change", { target: { value } });
        expect(wrapper.find("p").text().trim()).toBe(value);
    });

    test("Should not post the info with submit", () => {
        wrapper.find("form").simulate("submit", { preventDefault() { } });

        expect(setCategories).not.toHaveBeenCalled();
    });

    test("Should call setCategories and clean the textbox", () => {
        // 1. Simular el inputChange
        // 2. Simular el submit
        // 3. setCategories se debe de haber llamado
        // 4. el valor del input debe de estar ""
        const value = "Hola mundo desde clean textbox";
        wrapper.find("input").simulate("change", { target: { value } });
        wrapper.find("form").simulate("submit", { preventDefault() { } });
        /* let input = wrapper.find("input");
        let form = wrapper.find("form");
        input.simulate("change", { target: { value } });
        form.find("form").simulate("submit", { preventDefault() { } }); */
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        expect(wrapper.find("input").prop("value")).toBe("");
    });
});