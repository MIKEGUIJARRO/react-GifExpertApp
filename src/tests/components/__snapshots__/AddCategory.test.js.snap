// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory /> test Shows the <AddCategory /> correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
